src/components/MultiHopTrade/hooks/useGetTradeQuotes/useGetTradeRates.tsx

```diff
@@ -0,0 +1,290 @@
+    // and losing sync on loading state etc.
+    dispatch(tradeQuoteSlice.actions.clear())
+
+    // Early exit on any invalid state
+    if (bnOrZero(sellAmountCryptoPrecision).isZero()) {
+      setTradeRateInput(skipToken)
+      dispatch(tradeQuoteSlice.actions.setIsTradeQuoteRequestAborted(true))
+      return
+    }
+    ;(async () => {
+      const sellAccountNumber = sellAccountMetadata?.bip44Params?.accountNumber
+
+      const tradeAmountUsd = bnOrZero(sellAssetUsdRate).times(sellAmountCryptoPrecision)
+
+      const { feeBps, feeBpsBeforeDiscount } = calculateFees({
+        tradeAmountUsd,
+        foxHeld: bnOrZero(votingPower),
+        thorHeld: bnOrZero(thorVotingPower),
+        feeModel: 'SWAPPER',
+      })
+
+      const potentialAffiliateBps = feeBpsBeforeDiscount.toFixed(0)
+      const affiliateBps = feeBps.toFixed(0)
+
+      const updatedTradeRateInput = (await getTradeQuoteInput({
+        sellAsset,
+        sellAccountNumber,
+        sellAccountType: sellAccountMetadata?.accountType,
+        buyAsset,
+        wallet: wallet ?? undefined,
+        quoteOrRate: 'rate',
+        receiveAddress: undefined,
+        sellAmountBeforeFeesCryptoPrecision: sellAmountCryptoPrecision,
+        allowMultiHop: true,
+        affiliateBps,
+        potentialAffiliateBps,
+        // Pass in the user's slippage preference if it's set, else let the swapper use its default
+        slippageTolerancePercentageDecimal: userSlippageTolerancePercentageDecimal,
+        pubKey:
+          wallet && isLedger(wallet) && sellAccountId
+            ? fromAccountId(sellAccountId).account
+            : undefined,
+      })) as GetTradeRateInput
+
+      setTradeRateInput(updatedTradeRateInput)
+    })()
+  }, [
+    buyAsset,
+    dispatch,
+    sellAccountMetadata,
+    sellAmountCryptoPrecision,
+    sellAsset,
+    votingPower,
+    thorVotingPower,
+    wallet,
+    receiveAccountMetadata?.bip44Params,
+    userSlippageTolerancePercentageDecimal,
+    sellAssetUsdRate,
+    sellAccountId,
+    isVotingPowerLoading,
+    isBuyAssetChainSupported,
+  ])
+
+  const getTradeQuoteArgs = useCallback(
+    (swapperName: SwapperName): UseGetSwapperTradeQuoteArgs => {
+      return {
+        swapperName,
+        tradeQuoteInput: tradeRateInput,
+        skip: !shouldRefetchTradeQuotes,
+        pollingInterval:
+          swappers[swapperName]?.pollingInterval ?? DEFAULT_GET_TRADE_QUOTE_POLLING_INTERVAL,
+      }
+    },
+    [shouldRefetchTradeQuotes, tradeRateInput],
+  )
+
+  useGetSwapperTradeQuote(getTradeQuoteArgs(SwapperName.CowSwap))
+  useGetSwapperTradeQuote(getTradeQuoteArgs(SwapperName.ArbitrumBridge))
+  useGetSwapperTradeQuote(getTradeQuoteArgs(SwapperName.Portals))
+  useGetSwapperTradeQuote(getTradeQuoteArgs(SwapperName.LIFI))
+  useGetSwapperTradeQuote(getTradeQuoteArgs(SwapperName.Thorchain))
+  useGetSwapperTradeQuote(getTradeQuoteArgs(SwapperName.Zrx))
```

Comment:
It feels like we might want to put this into a util, `useFetchAllSwapperTradeQuotes` or something.

Reply:
100% agree with this guy conceptually but since the `getTradeQuoteArgs` logic is actually highly dependant on this hook's state (and is different for each swapper), this would require a major refactor to move all the logic to get to this input away too, would rather keep it as it was previously in develop's `useGetTradeQuotes.tsx` 🙏🏽